doc/api/buffer.md

```diff
@@ -71,6 +71,18 @@ first argument, security and reliability issues can be inadvertently introduced
 into applications when argument validation or `Buffer` initialization is not
 performed.
 
+For example, if an application can be convinced to call `new Buffer(100)`
+instead of `new Buffer("100")`, it will allocate a 100 byte buffer instead
+of allocating a 3 byte buffer with content `"100"`. This is commonly possible
+using JSON API calls. Since JSON distinguishes between numeric and string types,
+it allows injection of numbers where a naive application might expect to always
+receive a string.  Before Node.js 8.0.0, the 100 byte buffer might contain
+arbitrary pre-existing in-memory data, so may be used to expose in-memory
+secrets to a remote attacker.  Since Node.js 8.0.0, exposure of memory can
+not occur because the data is zero-filled. However, other attacks are still
```

Comment:
I wonder if it's really an advantage to include the description about old Node.js versions. Since the documentation is meant for newer versions, it's probably best to limit the description to the remaining attack vectors?

Reply:
If you expand the context diff upwards, you will see that this information is in an entire section dedicated to describing the situation in 6.x: how it used to work, why it was a problem, and guidelines on how to update from code developed pre-8.x to the new APIs.